pkg/pipeline/pipeline_test.go

```diff
@@ -104,4 +91,49 @@ func Test_SimplePipeline(t *testing.T) {
 	writer := mainPipeline.Writer.(*write.WriteNone)
 	require.Equal(t, len(ingester.PrevRecords), len(decoder.PrevRecords))
 	require.Equal(t, len(ingester.PrevRecords), len(writer.PrevRecords))
+
+	// checking that the processing is done for at least the first line of the logs
+	require.Equal(t, ingester.PrevRecords[0], decoder.PrevRecords[0])
+	// values checked from the first line of the ../../hack/examples/ocp-ipfix-flowlogs.json file
+	require.Equal(t, config.GenericMap{
+		"fl2m_bytes":   float64(20800),
+		"fl2m_dstAddr": "10.130.2.2",
+		"fl2m_dstPort": float64(36936),
+		"fl2m_packets": float64(400),
+		"fl2m_srcAddr": "10.130.2.13",
+		"fl2m_srcPort": float64(3100),
+	}, writer.PrevRecords[0])
+}
+
+func BenchmarkPipeline(b *testing.B) {
+	t := &testing.T{}
+	loadGlobalConfig(t)
```

Comment:
@mariomac can you look on https://github.com/netobserv/flowlogs-pipeline/blob/main/Makefile#L88 --- maybe we can somehow improve this >>> ???? I started to create some benchmark area for the project and I totally agree we need to improve that

Reply:
@mariomac In any event, can we agree to create dedicated go files just for benchmark and split from the rest of the tests so we can run those stand-alone??

Reply:
With respect to your second question, we could do that if you prefer it. Anyway benchmarks are not run by default even if they are in the same file as the tests. If you mean skipping tests when you run benchmarks, you can add the `-test.run=^$` argument to the `go test` command to skip any test.

But I'm fine if you feel it's better organizing everyting in the same benchmarks file.

Reply:
@eranra I see what's happening with the benchmark in `benchmark_test.go`. It basically runs many times the whole command, so it would measure basically the performance of starting the FL2M process and finishing it, not the actual FL2M processing performance.

Go benchmarks are more a sort of "micro-benchmarks" aimed to test some parts of the code, and that's why they are usually located in the test files of the components that they are benchmarking. For example, in the benchmark of this PR, it just tests the time of sending and processing a file of ~5000 flows with a very simple dummy pipeline (no real ingest, no real writing...), but it allows us measuring the impact of the sequential vs parallel pipeline mechanism.

I'd suggest to (in another PR to not loose the focus of our current task) remove the current `benchmark_test.go` and prepare some benchmark that:

1. starts the FL2M service with a real ingester
2. spins up a client that is able to send real IPFIX flows ([we did one using a VM library in our goflow-kube](https://github.com/netobserv/goflow2-kube-enricher/blob/24fc6050d63bfb913269af4e8675cdf66ceef310/test/integration/service.go#L124-L160))
3. Measures how many flows it's able to process in a given amount of time. E.g. the last stage of the pipeline could be just a counter.

In the future, this could be improved e.g. spinning parallel clients